kubespawner/spawner.py

```diff
@@ -1338,11 +1376,23 @@ def _validate_image_pull_secrets(self, proposal):
                 "command": ["/usr/local/bin/supercronic", "/etc/crontab"]
             }]
 
-        `{username}`, `{userid}`, `{servername}`, `{hubnamespace}`,
-        `{unescaped_username}`, and `{unescaped_servername}` will be expanded if
-        found within strings of this configuration. The username and servername
-        come escaped to follow the `DNS label standard
-        <https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names>`__.
+        .. seealso::
+
+          :ref:`templates` for information on fields available in template strings.
+
+        """,
+    )
+
+    handle_legacy_names = Bool(
+        True,
+        config=True,
+        help="""handle legacy names and labels
+        
+        kubespawner 7 changed the scheme for computing names and labels to be more reliably valid.
+        In order to preserve backward compatibility, the old names must be handled in some places.
+
+        You can safely disable this if no PVCs were created or running servers were started
+        before upgrading to kubespawner 7.
```

Comment:
This is only used to handle a transition to safe name expansions for PVCs, not for anything else like the leading help text `handle legacy names and labels` indicates it may.

I think updating this to describe its just about managing the PVC naming makes sense rather than trying to handle anything beyond the PVC naming.

Reply:
Yes, I intended to have this for a catch-all config "check if the upgraded defaults might have detached something without user input," but PVCs were the only thing I found where there was something to be done. Maybe namespace should be handled as well?

Reply:
namespaces don't need to be handled because they were persisted in 6.x, so I think this is okay as-is.